scripts/npm-safe-publish.sh

```diff
@@ -86,8 +93,29 @@ if [[ "$NPM_SHOW" != "$NPM_LS" ]]; then
       -d environment=production \
       -d sha=$SHA https://app.sleuth.io/api/1/deployments/taplytics/js-sdks-2/register_deploy
   else
+    echo "NPM Publish Local"
     npm publish --local
   fi
 else
   echo "Versions are the same ($NPM_SHOW = $NPM_LS). Not pushing"
 fi
+
+# If DEPRECATED_PACKAGE is set, run the deploy logic for it
+if [[ "$DEPRECATED_PACKAGE" != "" ]]; then
+  echo "Deploy to Deprecated Package: $DEPRECATED_PACKAGE"
+
+  # Backup the original package.json
+  cp package.json package.json.bak
+
+  # Update the name field to DEPRECATED_PACKAGE
+  jq --arg DEPRECATED_PACKAGE "$DEPRECATED_PACKAGE" ".name = $DEPRECATED_PACKAGE" package.json > package.json.temp
+  mv package.json.temp package.json
+
+  # Deploy logic
+  # Replace this with your actual npm publish command for DEPRECATED_PACKAGE
+  echo "Publishing $DEPRECATED_PACKAGE@$NPM_LS to NPM."
+  npm publish --otp=$OTP
+
+  # Restore the original package.json
+  mv package.json.bak package.json
```

Comment:
any way to make sure this happens whether or not npm publish succeeds or exits?

Reply:
will ask chatGPT